lib/Core/Inference/FrameBuilder/VariableWalker.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+namespace Phpactor\WorseReflection\Core\Inference\FrameBuilder;
+
+use Microsoft\PhpParser\Token;
+use Phpactor\WorseReflection\Core\Inference\Symbol;
+use Phpactor\WorseReflection\Core\Inference\Variable as WorseVariable;
+use Microsoft\PhpParser\Node;
+use Microsoft\PhpParser\Node\Expression\Variable;
+use Phpactor\WorseReflection\Core\Inference\Frame;
+use Phpactor\WorseReflection\Core\Inference\SymbolFactory;
+use Phpactor\WorseReflection\Core\DocBlock\DocBlockFactory;
+use Phpactor\WorseReflection\Core\Inference\FullyQualifiedNameResolver;
+use Phpactor\WorseReflection\Core\Inference\FrameBuilder;
+
+class VariableWalker
+{
+    /**
+     * @var DocBlockFactory
+     */
+    private $docblockFactory;
+
+    /**
+     * @var array
+     */
+    private $injectedTypes = [];
+
+    /**
+     * @var SymbolFactory
+     */
+    private $symbolFactory;
+
+    /**
+     * @var FullyQualifiedNameResolver
+     */
+    private $nameResolver;
+
+    public function __construct(
+        SymbolFactory $symbolFactory,
+        DocBlockFactory $docblockFactory,
+        FullyQualifiedNameResolver $nameResolver
+    )
+    {
+        $this->symbolFactory = $symbolFactory;
+        $this->docblockFactory = $docblockFactory;
+        $this->nameResolver = $nameResolver;
+    }
+
+    public function canWalk(Node $node)
+    {
+        return true;
+    }
+
+    public function walk(FrameBuilder $builder, Frame $frame, Node $node)
+    {
+        $this->injectVariablesFromComment($frame, $node);
```

Comment:
?? We need to check if variable nodes have comments overriding their types, but f.e. `foreach` statements can also have these phpdocs "injecting" a type for their items, so we catch all "injected" vars here...